### DISCUSSION TITLE
How to negotiate transport protocols?

### DISCUSSION THREAD

**Discussion by kbData** [eclipse-edc/Connector#781] [Q&A]
in case if multiple transport protocols are supported by connectors, how is a transport protocol for the transfer agreed upon between the communication parties? can consumer or producer put prios (i.e S3>FTP)? is the tranport protocol negotiation reflected in the APIs as of now? is there a must-have set of transport protocols which has to be implemented by any connector on the network to ensure communication (i.e. FTP as an ultimate fallback?)

**Comment by paullatzelsperger** [comment]
Consumers may specify how they want to receive the data, meaning, they can specify the data destination. This also includes the destination _type_, e.g. "AmazonS3" or "FTP" or whatever. 

Of course as a precondition, both the consumer and the provider must implement the same binding. The connector may announce during self-description what bindings it supports.
In case two connectors don't have any common bindings, we are thinking about a fallback solution, most likely that's gonna be HTTP in one form or another.

**Comment by jimmarino** [comment]
@paullatzelsperger @juliapampus @ronjaquensel this brings up a good point with respect to ContractOffers. The asset will need to convey the transfer protocols available. This could be on a per-asset or group basis. Something to discuss post-Milestone 1.

**Comment by paullatzelsperger** [comment]
converting this to a discussion since there is no clear assignment.
